src/src/lms/lms.service.ts

```diff
@@ -419,5 +379,428 @@ export class LMSService {
+		const userForIssueCertificate = await this.fetchTestTrackingData(
+			parseInt(
+				this.configService.get<string>(
+					'LMS_CERTIFICATE_ISSUE_BATCH_SIZE',
+				),
+			),
+		);
+		if (userForIssueCertificate.length > 0) {
+			for (let i = 0; i < userForIssueCertificate.length; i++) {
+				let userTestData = userForIssueCertificate[i];
+				let issue_status = '';
+				let minPercentage = parseFloat(
+					this.configService.get<string>(
+						'LMS_CERTIFICATE_ISSUE_MIN_SCORE',
+					),
+				);
+				let user_id = userTestData?.user_id;
+				let test_id = userTestData?.test_id;
+				let context = userTestData?.context;
+				let context_id = userTestData?.context_id;
+				let getUserList = await this.userService.getUserName(user_id);
+				let user_name = '';
+				if (getUserList.length > 0) {
+					/*let first_name = '';
+					let middle_name = '';
+					let last_name = '';
+					let temp = 'rushi gawali';
+					let temp_r = await this.CapitalizeEachWord(temp);*/
+
+					user_name += getUserList[0]?.first_name
+						? (await this.CapitalizeEachWord(
+								getUserList[0].first_name,
+						  )) + ' '
+						: '';
+					user_name += getUserList[0]?.middle_name
+						? (await this.CapitalizeEachWord(
+								getUserList[0].middle_name,
+						  )) + ' '
+						: '';
+					user_name += getUserList[0]?.last_name
+						? (await this.CapitalizeEachWord(
+								getUserList[0].last_name,
+						  )) + ' '
+						: '';
+				}
+				//get attendance status
+				let attendance_valid = false;
+				let usrAttendanceList =
+					await this.attendanceCoreService.getUserAttendancePresentList(
+						user_id,
+						context,
+						context_id,
+					);
+				let minAttendance = parseInt(
+					this.configService.get<string>(
+						'LMS_CERTIFICATE_ISSUE_MIN_ATTENDANCE',
+					),
+				);
+				if (usrAttendanceList.length >= minAttendance) {
+					attendance_valid = true;
+				}
+				//check certificate criteria
+				if (userTestData?.score >= minPercentage && attendance_valid) {
+					issue_status = 'true';
+				} else {
+					issue_status = 'false';
+				}
+				//issue certificate
+				if (issue_status == 'true') {
+					let temp_html_code = html_code;
+					let issuance_date = moment().format('YYYY-MM-DD');
+					let issuance_date_tx = moment().format('DD MMM YYYY');
+					let expiration_date = moment(issuance_date)
+						.add(12, 'M')
+						.format('YYYY-MM-DD');
+					const lmsCertificate = new LMSCertificateDto({
+						user_id: user_id,
+						test_id: test_id,
+						certificate_status: 'Issued',
+						issuance_date: issuance_date,
+						expiration_date: expiration_date,
+					});
+					const certificate_data = await this.issueCertificateHtml(
+						lmsCertificate,
+					);
+					if (certificate_data != null) {
+						let certificate_id = certificate_data?.id;
+						let uid = 'P-' + certificate_id + '-' + user_id;
+						//update html code
+						temp_html_code = temp_html_code.replace(
```

Comment:
@rushi-tekdi Change this variable name to certificateTemplate

Reply:
changed